python/lsst/analysis/drp/gatherResourceStatistics.py

```diff
@@ -172,19 +180,153 @@ class GatherResourceStatisticsTask(PipelineTask):
+            }
+            if task_def.metadataDatasetName is not None:
+                output_metadata_dataset_type = DatasetType(
+                    task_def.metadataDatasetName,
+                    dimensions=empty_dimensions,
+                    storageClass="TaskMetadata",
+                )
+                outputs[output_metadata_dataset_type] = [DatasetRef(output_metadata_dataset_type, data_id)]
+            if task_def.logOutputDatasetName is not None:
+                log_dataset_type = DatasetType(
+                    task_def.logOutputDatasetName,
+                    dimensions=empty_dimensions,
+                    storageClass="ButlerLogRecords",
+                )
+                outputs[log_dataset_type] = [DatasetRef(log_dataset_type, data_id)]
+            quantum = Quantum(
+                taskName=task_def.taskName,
+                taskClass=task_def.taskClass,
+                dataId=data_id,
+                inputs={input_metadata_dataset_type: list(metadata_refs)},
+                outputs=outputs,
+            )
+            quanta_by_task_def[task_def] = {quantum}
+        return QuantumGraph(quanta_by_task_def)
+
+    @classmethod
+    def build_quantum_graph_cli(cls, argv):
+        """A command-line interface entry point to `build_quantum_graph`.
+
+        This method provides the implementation for the
+        ``build-gather-resource-usage-qg`` script.
+
+        Parameters
+        ----------
+        argv : `Sequence` [ `str` ]
+            Command-line arguments (e.g. ``sys.argv[1:]``).
+        """
+        parser = argparse.ArgumentParser(
+            description=(
+                "Build a QuantumGraph that runs GatherResourceStatisticsTask on existing metadata datasets."
+            ),
+        )
+        parser.add_argument("repo", type=str, help="Path to data repository or butler configuration.")
+        parser.add_argument("filename", type=str, help="Output filename for QuantumGraph.")
+        parser.add_argument(
+            "collections",
+            type=str, nargs="+",
+            help="Collection(s)s to search for input metadata.",
+        )
+        parser.add_argument(
+            "--dataset-types",
+            type=str,
+            action="extend",
+            help=(
+                "Glob-style patterns for input metadata dataset types.  If a pattern matches a "
+                "non-metadata dataset type, non-metadata matches will be ignored."
+            ),
+        )
+        parser.add_argument(
+            "--where",
+            type=str,
+            default=None,
+            help="Data ID expression used when querying for input metadata datasets.",
+        )
+        args = parser.parse_args(argv)
+        base_dataset_type_filter = re.compile(r"\w+_metadata")
+        if not args.dataset_types:
+            input_dataset_types = base_dataset_type_filter
+        else:
+            input_dataset_types = [fnmatch.translate(expr) for expr in args.dataset_types]
```

Comment:
As a note, we wrote `lsst.daf.butler.core.utils.gloToRegex` to handle this in our command line tooling so you might want to use that here to ensure consistency (and it has the advantage that if you pass in a non-regex it returns back the string and not a regex).